File: src/tact-and-led.pl
#!/usr/bin/env perl

use strict;
use warnings;
use utf8;
use Carp;
use 5.020;
use IO::File;
use AnyEvent::RaspberryPi::GPIO;
use AnyEvent;

main();

sub main {
  my $cv     = AE::cv;
  my $gpio17 = AnyEvent::RaspberryPi::GPIO->new({
    channel   => 17,
    direction => "out",
    verbose   => 1,
    onchange  => sub {

    }
  });

  my $gpio4 = AnyEvent::RaspberryPi::GPIO->new({
      channel   => 4,
      direction => "in",
      verbose   => 1,
      onchange  => sub {
        my $value = shift;
        $gpio17->set($value);
      }
    });

  $cv->begin;
  my $finalize = AE::signal "INT", sub {
    $cv->end;
  };

  $cv->recv;
  #$gpio4->unexport();
  #$gpio17->unexport();
  say "\nBye!!";
}

File: src/tact-and-led.js
"use strict";

const gpio = require("gpio");

const channel = {
  tact: 4,
  led: 17
};

const gpioTact = gpio.export(channel.tact, {
  direction: "in",
  ready(){
    console.log("GPIO4 is ready!!");
  }
});

const gpioLed = gpio.export(channel.led, {
  direction: "out",
  ready(){
    console.log("GPIO17 is ready!!");
  }
});

gpioTact.on("change", (val) => {
  console.log(val);
  if(val === 1){
    gpioLed.set();
  }else{
    gpioLed.set(0);
  }
});

process.on("SIGINT", () => {
  gpioLed.reset();
  gpioLed.unexport();
  gpioTact.unexport();
  console.log("\nBye!!");
});
